marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/entity-lib.sjs

```diff
@@ -0,0 +1,140 @@
+  const map = {};
+  for (var doc of cts.search(cts.collectionQuery("http://marklogic.com/entity-services/models"))) {
+    Object.assign(map, convertModelToEntityTypeMap(doc.toObject()));
+  }
+  return map;
+}
+
+/**
+ * @param entityTypeId string or a sem.iri
+ * @return the matching model document, or null if one is not found
+ */
+function findModelForEntityTypeId(entityTypeId) {
+  return fn.head(cts.search(
+    cts.andQuery([
+      cts.collectionQuery('http://marklogic.com/entity-services/models'),
+      cts.tripleRangeQuery(sem.iri(entityTypeId), sem.curieExpand("rdf:type"), sem.curieExpand("es:EntityType", semPrefixes))
+    ])));
+}
+
+/**
+ * @param entityTypeId string or a sem.iri
+ * @return null if a model can't be found matching the given EntityTypeId, or if a model is found but there's no an entity type
+ * with a title matching the entity title in the EntityTypeId. Otherwise, the entity type from the definitions array in the model
+ * is returned.
+ */
+function findEntityType(entityTypeId) {
+  const modelDoc = findModelForEntityTypeId(entityTypeId);
+  if (!modelDoc) {
+    return null;
+  }
+  return modelDoc.toObject().definitions[getEntityTypeIdParts(entityTypeId).entityTypeTitle];
+}
+
+/**
+ * @param entityTypeId sem.iri or string
+ * @return {{entityTypeTitle: string, baseUri: string, modelTitle: string, version: string}}
+ */
+function getEntityTypeIdParts(entityTypeId) {
+  if (entityTypeId == null) {
+    throw Error("Cannot get EntityTypeId parts from null EntityTypeId");
+  }
+
+  entityTypeId = entityTypeId.toString();
+
+  const tokens = entityTypeId.split("/");
+  if (tokens.length < 3) {
+    throw Error("Could not get EntityTypeId parts from invalid EntityTypeId: " + entityTypeId);
+  }
+
+  const infoTokens = tokens[tokens.length - 2].split("-");
+  if (infoTokens.length < 2) {
+    throw Error("Could not get EntityTypeId parts; expected info part did not contain a hyphen; EntityTypeId: " + entityTypeId);
+  }
+
+  const title = tokens[tokens.length - 1];
+  const baseUri = tokens.slice(0, tokens.length - 2).join("/") + "/";
+  return {
+    baseUri: baseUri,
+    modelTitle: infoTokens[0],
+    version: infoTokens.slice(1).join("-"),
+    entityTypeTitle: title
+  };
+}
+
+/**
+ * @param model the model object as found in a model descriptor
+ * @param entityTypeTitle a string identifying the entity type in the definitions array that an IRI is needed for
+ * @return {string}
+ */
+function getEntityTypeId(model, entityTypeTitle) {
+  return getModelId(model) + "/" + entityTypeTitle;
+}
+
+/**
+ * @param model
+ * @return a map (object) where each key is an EntityTypeId and the value is the EntityType
+ */
+function convertModelToEntityTypeMap(model) {
+  const map = {};
+  const modelId = getModelId(model);
+  for (var entityTypeTitle of Object.keys(model.definitions)) {
+    map[modelId + "/" + entityTypeTitle] = model.definitions[entityTypeTitle];
+  }
+  return map;
+}
+
+function getModelId(model) {
```

Comment:
for consistency, add java doc?

Reply:
I didn't yet because it's not an exported function. We also don't have any expectation of comments on methods - the vast majority of DH functions don't have comments. I only included them where I felt they added value in this library, and I didn't think they did for getModelId, which is simple enough to understand by looking at the code.